src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -182,11 +189,14 @@ module.exports = function<T, P, I, TI, C>(
     } else {
       shouldUpdate = updateClassInstance(current, workInProgress);
     }
+    const instance = workInProgress.stateNode;
+    if (typeof instance.getChildContext === 'function') {
+      pushContextProvider(workInProgress);
```

Comment:
I should probably do this after rendering. Otherwise context provider would likely see its own context.